lang/ghc/Portfile

```diff
@@ -29,144 +24,145 @@ long_description    \
+    if {[string trim ${gpg_verification}] != "VERIFIED"} {
+        ui_error "GPG signature verification failed with keyid ${keyid}."
     }
 }
 
-post-activate {
-    set libprefix "${prefix}/lib/${name}-${version}"
-
-    # delete old ${prefix}/lib/ghc-${version} directories
-    foreach fullpath [glob -nocomplain -directory ${prefix}/lib ghc-*] {
-        if {$fullpath ne $libprefix} {
-            ui_info "Cleaning up remnant GHC library directory ${fullpath}"
-            catch {delete $fullpath}
+extract.only        ${name}-${version}-x86_64-apple-darwin${extract.suffix}
+
+# fix ${buildmakejobs} to be hw.physicalcpu+1 
+# https://github.com/macports/macports-base/blob/4b36e605d76fa3a423b4e35a6722ade4185ffa4f/src/port1.0/portbuild.tcl#L137
+namespace path { ::tcl::mathop }
+global buildmakejobs
+set buildmakejobs   [+ [sysctl hw.physicalcpu] 1]
+
+configure.args      --prefix=${destroot}${prefix}
+
+build {}
+
+destroot {
+    system -W ${worksrcpath} "make install -j${buildmakejobs}"
+
+    if {[variant_isset "bootstrap"]} {
+        # bootstrap build/destroot from ${name}-${version}-src${extract.suffix}
+        # the PATH environment must provide /usr/bin/gcc, not ${prefix}/bin/gcc
+        xinstall -m 0755 -d ${workpath}/src
+        system -W ${workpath}/src \
+            "tar xJf ${distpath}/${name}-${version}-src${extract.suffix}"
+        # fix DejaVu texlive fonts names, e.g. "\setmonofont{DejaVu Sans Mono}"
+        foreach f "\
+            ${workpath}/src/${name}-${version}/libraries/Cabal/Cabal/doc/conf.py \
+            ${workpath}/src/${name}-${version}/docs/users_guide/conf.py \
+            " {
+            reinplace -E "s|(\{DejaVu\[\[:alnum:\]_\]*)\[\[:space:\]\]+|\\1|g" \
+                ${f}
+            reinplace -E "s|(\{DejaVu\[\[:alnum:\]_\]*)\[\[:space:\]\]+|\\1|g" \
+                ${f}
+            reinplace -E "s|\{(DejaVu\[\[:alnum:\]_\]*)\}|\{\\1.ttf\}|g" \
+                ${f}
+        }
+        # fix python3 executable name
+        foreach f "\
+            ${workpath}/src/${name}-${version}/boot \
+            ${workpath}/src/${name}-${version}/utils/checkUniques/check-uniques.py \
+            " {
+            reinplace -E "s|^(#!/usr/bin/env python)3|\\1${python3_version}|" \
+                ${f}
         }
+        system -W ${workpath}/src/${name}-${version} \
+            "export PATH=/usr/bin:\${PATH} CC=/usr/bin/gcc ; \
+                ./boot"
+        system -W ${workpath}/src/${name}-${version} \
+            "export PATH=/usr/bin:\${PATH}:${destroot}${prefix}/bin CC=/usr/bin/gcc ; \
+                ./configure --prefix=${destroot}${prefix}"
+        system -W ${workpath}/src/${name}-${version} \
+            "export PATH=/usr/bin:\${PATH} CC=/usr/bin/gcc ; \
+                make -j${buildmakejobs}"
+        system -W ${workpath}/src/${name}-${version} \
+            "export PATH=/usr/bin:\${PATH} CC=/usr/bin/gcc ; \
+                make install -j${buildmakejobs}"
```

Comment:
@mojca [said](https://github.com/macports/macports-ports/pull/4699#discussion_r298580834) the same thing but…
* I have no idea what a support is or how they work and `man portfile` doesn't illuminate the issue.
* I still believe this is true:
> With a separate port, I’ll have to worry about destrooting the 1st stage, installing over it, communicating between two ports via the file system, having two ports control the same file, and other stuff. That sounds a lot more complicated than just running a few build command in an if block, as currently proposed.

In truth, I have no idea how to implement bootstrapping in MacPorts, and don't have the resources to get it all working.

What I can do is just run the build commands that the upstream package says to run, which are implemented in this Portfile.

Reply:
The is out of scope.

If anyone ever implements a bootstrap phase in MacPorts or a first stage port, please design it so that the first stage isn't kept around.

Reply:
I replaced explicit build commands with calls to MacPorts functions: `portextract::extract_main`, `portconfigure::configure_main`, etc.

If you search for `bootstrap` in MacPorts, you'll see that the approach taken in here is used in several other ports.